rs/execution_environment/tests/canister_history.rs

```diff
@@ -1257,6 +1257,117 @@ fn check_environment_variables_for_create_canister_history(
     }
 }
 
+#[test]
+fn canister_history_tracking_env_vars_update_settings() {
+    let user_id1 = user_test_id(7).get();
+    let user_id2 = user_test_id(8).get();
+
+    let env_vars = BTreeMap::from([
+        ("NODE_ENV".to_string(), "production".to_string()),
+        ("LOG_LEVEL".to_string(), "info".to_string()),
+    ]);
+
+    // Set up StateMachine.
+    let anonymous_user = PrincipalId::new_anonymous();
+    let env = setup_state_machine(FlagStatus::Enabled);
+
+    // Set time of StateMachine to current system time.
+    let mut now = std::time::SystemTime::now();
+    env.set_time(now);
+
+    // Create and install universal_canister.
+    let ucan = env
+        .install_canister_with_cycles(
+            UNIVERSAL_CANISTER_WASM.to_vec(),
+            vec![],
+            Some(
+                CanisterSettingsArgsBuilder::new()
+                    .with_controllers(vec![anonymous_user, user_id1, user_id2])
+                    .build(),
+            ),
+            INITIAL_CYCLES_BALANCE * 3_u64,
+        )
+        .unwrap();
+
+    // Create a canister with environment variables.
+    now += Duration::from_secs(5);
+    env.set_time(now);
+    let payload = CreateCanisterArgs {
+        settings: Some(
+            CanisterSettingsArgsBuilder::new()
+                .with_controllers(vec![user_id1, user_id2])
+                .with_environment_variables(env_vars.clone())
+                .build(),
+        ),
+        sender_canister_version: Some(2),
+    }
+    .encode();
+    let ucan_payload = universal_canister_payload(
+        &PrincipalId::default(),
+        "create_canister",
+        payload,
+        INITIAL_CYCLES_BALANCE * 2_u64,
+    );
+    let wasm_result = env.execute_ingress(ucan, "update", ucan_payload).unwrap();
+    let canister_id = canister_id_from_wasm_result(wasm_result);
```

Comment:
Canister creation is tested separately (in the following two tests) so I'd suggest to not repeat it in this test.